debug_module.tex

```diff
@@ -382,9 +382,18 @@ \section{Security}
 specific, it is not further addressed in this spec.
 
 Another option is to allow the DM to be unlocked only by users who have an
-access key. A few bits in \Rdmstatus and \Rauthdata can support an arbitrarily
-complex authentication mechanism.  When \Fauthenticated is clear, the DM must
-not interact with the rest of the platform in any way.
+access key. Between \Fauthenticated, \Fauthbusy, and \Rauthdata arbitrarily
+complex authentication mechanism can be supported.  When \Fauthenticated is
+clear, the DM must not interact with the rest of the platform, nor expose
+details about the harts connected to the DM. All DM registers should read 0,
+while writes should be ignored, with the following mandatory exceptions:
+\begin{steps}{}
+    \item \Fauthenticated in \Rdmstatus is readable.
+    \item \Fauthbusy in \Rdmstatus is readable.
+    \item \Fversion in \Rdmstatus is readable.
+    \item \Fdmactive in \Rdmcontrol is readable and writable.
+    \item \Rauthdata is readable and writable.
+\end{steps}
 
```

Comment:
This looks good... one other question, is \Fauthenticated = 0 similar to a reset/dmactive = 0? I mean, does clearing the authentication reset the other debug module registers (or *might* it)?

Reply:
No. I think of authentication as orthogonal to DM reset, or any other DM operation. If you're authenticated you can perform operations on the DM. If you're not authenticated then you cannot. In general I feel things don't have side effects unless they are explicitly mentioned. Do you think it's helpful to add:

DM reset is completely independent of DM authentication.

Reply:
I ask because I thought we said something about authentication and reset elsewhere in the spec, but I may be mistaken. Let me grep a bit.